packages/components/pie-toggle-switch/src/toggle-switch.scss

```diff
@@ -1,3 +1,129 @@
-// TODO - please remove the stylelint rule above when SCSS is added to this file
-/* stylelint-disable no-empty-source */
+@mixin toggle-switch-transition ($property) {
+    @media (prefers-reduced-motion: no-preference) {
+        transition-property: $property;
+        transition-duration: 0.15s;
+        transition-timing-function: cubic-bezier(0.4, 0, 0.9, 1);
+        transition-delay: 0s;
+    }
+}
 
+*, *:before, *:after {
+    box-sizing: border-box;
+}
+
+.c-toggle-switch {
+    --toggle-switch-colour:              var(--dt-color-interactive-form);
+    --toggle-switch-colour--checked:     var(--dt-color-interactive-brand);
+    --toggle-switch-colour--disabled:    var(--dt-color-disabled-01);
+    --toggle-switch-width:               48px;
+    --toggle-switch-height:              24px;
+    --toggle-switch-control-size:        20px;
+    --toggle-switch-padding:             2px;
+    --toggle-switch-radius:              var(--dt-radius-rounded-e);
+    --toggle-switch-icon-size:           14px;
+
+    --toggle-switch-translation: calc(var(--toggle-switch-width) - var(--toggle-switch-control-size) - 2 * var(--toggle-switch-padding));
+
+    display: flex;
+    @include toggle-switch-transition(background-color);
+    width: var(--toggle-switch-width);
+    height: var(--toggle-switch-height);
+    flex-shrink: 0;
+    padding: var(--toggle-switch-padding);
+    border-radius: var(--toggle-switch-radius);
+    background-color: var(--toggle-switch-colour);
+    cursor: pointer;
+
+    &:hover {
+        background-color: hsl(var(--dt-color-interactive-form-h), var(--dt-color-interactive-form-s), calc(var(--dt-color-interactive-form-l) - var(--dt-color-hover-01)));
+    }
+
+    &:focus,
+    &:focus-within {
+        background-color: --toggle-switch-colour;
```

Comment:
nit think `--toggle-switch-colour` needs a var around it and maybe needs to be `color` instead? 😄